118_pascal's_triangle.py

```diff
@@ -0,0 +1,43 @@
+from typing import List
+
+
+class Solution:
+    def generate(self, numRows: int) -> List[List[int]]:
+
+        # Approach 1
+        # result = [[1] * i for i in range(1, numRows + 1)]
+        # for i in range(2, len(result)):
+        #     for j in range(1, i):
+        #         result[i][j] = result[i - 1][j - 1] + result[i - 1][j]
+        # return result
+
+        # Approach 2
+        """
+          // Time Complexity :  O(n^2)
+                            'n' is the number of rows
+          // Space Complexity : O(n^2)
+          // Did this code successfully run on Leetcode : Yes
```

Comment:
Why O(n2)? space

Reply:
For each value of 'n', we have corresponding number of elements. 
For n =5, 1+2+3+4+5 = (n * n+1)/2 = n^2. No?